app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VendorsController < ApplicationController
+  before_filter :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @vendors = Vendor.all
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
+    @items = @vendor.items
+  end
+
+  def new
+    @vendor = Vendor.new
+  end
+
+  def create
+    @vendor = Vendor.new(vendor_params)
+
+    if @vendor.save
+      redirect_to params[:next] || vendors_path, notice: "Vendor was successfully created"
+    else
+      render action: new
```

Comment:
nitpick: when this fails, I think the validation errors are not set in the flash?